homeassistant/components/synology_dsm/__init__.py

```diff
@@ -139,7 +159,8 @@ async def async_remove_config_entry_device(
     hass: HomeAssistant, entry: ConfigEntry, device_entry: dr.DeviceEntry
 ) -> bool:
     """Remove synology_dsm config entry from a device."""
-    data: SynologyDSMData = hass.data[DOMAIN][entry.unique_id]
+    assert entry.unique_id
```

Comment:
Most integrations use entry.entry_id for the key which avoids the need to assert

Reply:
I’d probably change it all in a separate PR and then update this one

Reply:
reverted in 9a26ccd42990ebf60cae1f4b269dbc57aaf2dcc9